### PR TITLE
CRM457-2198: Start using pricing gem

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
- Use the gem as the source of truth for pricing
- Use the gem as the source of truth for NSM calculations
- Because we don't want to have to trust calculated fields in the payload (the schema now marks calculated fields and pricing fields as deprecated), as they should always be recalculated dynamically, add a new `high_value` boolean attribute to the payload, as a half-way-house towards getting this stuff out entirely but also enabling the app store and caseworker app to filter in/out high value claims

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-2198)

### CODE REVIEW

File: app/forms/nsm/steps/letters_calls_form.rb
Comment (ivanELEC) [inline app/forms/nsm/steps/letters_calls_form.rb:42]: this might be me being dim and not seeing something obvious, but where is this letters method coming from?

Comment (patrick-laa) [reply app/forms/nsm/steps/letters_calls_form.rb:42]: It's an attribute defined on line 7. `letters_before_uplift` used to be nil of `letters` hadn't been set, but now it will be `0`, so to preserve existing behaviour I'm checking for `letters` being set here.

File: app/services/risk_assessment/high_risk_assessment.rb
Comment (ivanELEC) [inline app/services/risk_assessment/high_risk_assessment.rb:16]: Since this threshold is shared between the caseworker and provider apps - would it make sense to encorporate it into the common gem in the future?

Comment (patrick-laa) [reply app/services/risk_assessment/high_risk_assessment.rb:16]: Thoroughly agree! I've pitched doing exactly that in my "Move the biz logic around" ADR: https://dsdmoj.atlassian.net/wiki/spaces/CRM457/pages/5246943990/Draft+ADR+16+Business+logic+placement#%E2%9D%8C-D.-The-provider-app-is-currently-responsible-for-flagging-claims-as-high-risk

**Review by ivanELEC — COMMENTED** [review]
Just want to confirm my thinking re: the "steps" needed for applying the new library rather than look at the code in an isolated way: 

1. The LaaCrimeFormsCommon::Pricing module completely replaces the pricing library and therefore the library and config for the library is removed - any instance where we would have used the pricing library is replaced by it's respective method/attribute from the new common module

2. The claim model is updated so we get total values and rates as individual methods that can be used to replace the pricing module. The method's data_for_calculation and full_data_for_calculation are used to format data from the claim so that they can be input into the rates and totals method respectively.
3. A similar process for step 2 is applied to the disbursment_costs, letters_and_calls_costs and work_item_costs concerns. The rates method that is already in the claim is delegated so that it can be reused in the concerns. 
4. With these additional methods ready for the claim and individual costing items, we replace pricing calculations in the forms, presenters and views (also using some delegation to reuse methods from the model, rather than replicate logic) - also some additional cleanup on the views

**Review by ivanELEC — APPROVED** [review]

**Comment by ivanELEC** [comment]
Taking back the approval until the common gem is approved and merged and we've done the todo in the gemfile (to prevent us accidentally merging without doing that)

**Comment by elken** [comment]
You have to actually dismiss the review to undo it
image

**Comment by patrick-laa** [comment]
I've now removed the TODO

**Pull request merged**
